### PR TITLE
Polish C++ Wsman* classes

### PR DESCRIPTION
**Pull request by phatina**
Hi,

I did some polishing work in C++ bindings.

**What's new:**
- new class WsmanOptions
- refactored classes:
  - WsmanFilter
  - WsmanEPR
- simplified calls in OpenWsmanClient

Thanks for review.

### CODE REVIEW

**Comment by kkaempf** [comment]
Thanks a lot for giving the C++ bindings some love !

Please add a "License:" field to the header of newly added files.

**Comment by phatina** [comment]
Ah, thank you. Will do so.

**Comment by phatina** [comment]
**Things changed:**
- added BSD license to newly added files
- did some code cleanup in (Open)WsmanClient to make the code look more C++-ish

**Comment by kkaempf** [comment]
Thanks Peter.

However, "BSD" is not sufficient. Please make it "BSD-3-Clause" to match the Openwman license.

**Comment by phatina** [comment]
Changed to BSD-3-Clause.

**Comment by kkaempf** [comment]
Perfect. Thanks a lot for your contribution !

**Comment by wcewarrior** [comment]
These changes look great. Thanks Peter!

Klaus, are you planning any release soon that will include these changes (v2.4.13)?

File: src/cpp/OpenWsmanClient.h [hunk 1]
```diff
@@ -29,89 +30,147 @@ namespace WsmanClientNamespace
...
+			void SetProxy(
+				const string &proxy = string(),
+				const string &proxy_username = string(),
+				const string &proxy_password = string());
 #ifdef _WIN32
 			// Set client certificate params
-			void SetClientCert(const char *caOid=NULL, const char *caName=NULL, const bool localCert=false);
+			void SetClientCert(
+				const string &caOid = string(),
+				const string *caName = string(),
```
Comment (wcewarrior) [inline src/cpp/OpenWsmanClient.h]: This will not build on windows. Perhaps you meant this:
const string &caName = string(),

Comment (phatina) [inline src/cpp/OpenWsmanClient.h]: Yes, typo.

Comment (phatina) [inline src/cpp/OpenWsmanClient.h]: Please, have a look at:

https://github.com/Openwsman/openwsman/pull/56

**Pull request merged**